integrations/chroma/src/haystack_integrations/document_stores/chroma/filters.py

```diff
@@ -0,0 +1,145 @@
+from collections import defaultdict
+from typing import Any, Dict, List, Tuple, Union
+
+from chromadb.api.types import validate_where, validate_where_document
+
+from .errors import ChromaDocumentStoreFilterError
+
+OPERATORS = {
+    "==": "$eq",
+    "!=": "$ne",
+    ">": "$gt",
+    ">=": "$gte",
+    "<": "$lt",
+    "<=": "$lte",
+    "in": "$in",
+    "not in": "$nin",
+    "AND": "$and",
+    "OR": "$or",
+    "contains": "$contains",
+    "not contains": "$not_contains",
+}
+
+
+def _convert_filters(
+    filters: Union[Dict[str, Any], List[Dict[str, Any]]]
+) -> Tuple[List[str], Dict[str, Any], Dict[str, Any]]:
+    """
+    Converts Haystack filters into a format compatible with Chroma, separating them into ids, metadata filters,
+    and content filters to be passed to chroma as ids, where, and where_document clauses respectively.
+
+    """
+
+    ids = []
+    where: Dict[str, Any] = defaultdict(list)
+    where_document: Dict[str, Any] = defaultdict(list)
+
+    if isinstance(filters, dict):  # if filters is a dict, convert it to a list
+        filters = [filters]
+
+    for clause in filters:
+        normalized_clause = _normalize_filters(clause)
+        for field, value in normalized_clause.items():
+            if value is None:
+                continue
+            where_document.update(create_where_document_filter(field, value))
+            # if where_document is not empty, current clause is a content filter and we can skip rest of the loop
+            if where_document:
+                continue
+            # if field is "id", it'll be passed to Chroma's ids filter
+            elif field == "id":
+                if not value["$eq"]:
+                    msg = f"id filter only supports '==' operator, got {value}"
+                    raise ChromaDocumentStoreFilterError(msg)
+                ids.append(value["$eq"])
+            else:
+                where[field] = value
+
+    try:
+        if where_document:
+            test_clause = "document content filter"
+            validate_where_document(where_document)
+        elif where:
+            test_clause = "metadata filter"
+            validate_where(where)
+    except ValueError as e:
+        msg = f"Invalid '{test_clause}' : {e}"
+        raise ChromaDocumentStoreFilterError(msg) from e
+
+    return ids, where, where_document
+
+
+def _normalize_filters(filters: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Not sure if "normalize" is the right verb here - we're just converting one pre-normalized format to another. 

I think we can inline this function in `_convert_filters`:
```python
converted_clause = _parse_comparison_condition(clause) if "field" in clause else _parse_logical_condition(clause)
```

Reply:
Will cause a bit of repetition as this method used in line 117 of this file. Give a thumbs up if you still prefer to replace it.

Reply:
Ah, didn't notice that! We can keep it then, but we need a better name - there's no actual normalization going on, just conversion. `_convert_filter_clause` or something along those lines?